### PR TITLE
Added CI/CD actions for solutions

### PR DESCRIPTION
**Pull request by cdavernas**
- Adds CI/CD actions for solutions
- Adds issue and PR templates

### CODE REVIEW

File: .github/workflows/ci-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+jobs:
+  #test:
+  #  # Skip if it's not running in the main branch of the `neuroglia-io/hylo` repo
+  #  if: github.repository == 'neuroglia-io/hylo' && github.ref_name == 'main'
+  #  uses: ./.github/workflows/test-dotnet.yml
+  #  secrets: inherit
+
+  versioning:
+    #needs: [test]
+    if: github.repository == 'neuroglia-io/hylo' && github.ref_name == 'main' # comment if test are turned on
```
Comment (JBBianchi) [inline .github/workflows/ci-pipeline.yml]: This probably needs to be adapted if we want to turn tests ON before publishing

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        run: dotnet restore "${{ env.SOLUTION }}"
+      - name: Build
+        run: dotnet build "${{ env.SOLUTION }}" --configuration Release --no-restore 
+      - name: Push1
+        run: dotnet nuget push "./src/*/bin/Release/*.nupkg" --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json --skip-duplicate
+      - name: Push2
+        run: dotnet nuget push "./src/*/*/bin/Release/*.nupkg" --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json --skip-duplicate
+
+  publish-broker-bin:
+    name: Publish Broker Binaries
```
Comment (JBBianchi) [inline .github/workflows/publish.yml]: There is no broker

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          rm -r "$release_name"
+      - name: Publish
+        uses: softprops/action-gh-release@v1
+        with:
+          files: "cloud-streams-broker*"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  publish-gateway-bin:
+    name: Publish Gateway Binaries
```
Comment (JBBianchi) [inline .github/workflows/publish.yml]: There is no gateway

**Comment by JBBianchi** [comment]
There few issues:

- The publish action is not configured properly. It tries to build a broker and a gateway and set the ouput as GitHub release artifacts. 
- The repository is not configured properly:
  - [x] Neuroglia bot should be an admin of the repo
  - [x] Branch protection should be active for main
  - [x] Secrets are missing
- Tests are not run in the ci pipeline

File: .github/workflows/publish.yml (hunk 2 above)
Comment (cdavernas) [reply .github/workflows/publish.yml]: Yeah, I wanted to replace them with checkin that includes compiled and packaged Database Providers

File: .github/workflows/publish.yml (hunk 3 above)
Comment (cdavernas) [reply .github/workflows/publish.yml]: Same comment as above

File: .github/workflows/ci-pipeline.yml (hunk 1 above)
Comment (cdavernas) [reply .github/workflows/ci-pipeline.yml]: You want me to uncomment line 21 then?

Comment (JBBianchi) [reply .github/workflows/ci-pipeline.yml]: I think we can uncomment L14-L18 and L2,1 and comment L22, at least if we want to enforce tests before publishing.

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        with:
+          dotnet-version: 7.0.x
+      - name: Restore dependencies
+        run: dotnet restore "${{ env.SOLUTION }}"
+      - name: Build
+        run: dotnet build "${{ env.SOLUTION }}" --configuration Release --no-restore 
+      - name: Push1
+        run: dotnet nuget push "./src/*/bin/Release/*.nupkg" --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json --skip-duplicate
+      - name: Push2
+        run: dotnet nuget push "./src/*/*/bin/Release/*.nupkg" --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json --skip-duplicate
```
Comment (JBBianchi) [inline .github/workflows/publish.yml:28]: Missing secret `NUGET_API_KEY`

File: .github/workflows/publish.yml (hunk 2 above)
Comment (JBBianchi) [reply .github/workflows/publish.yml]: Could you elaborate please ?

File: .github/workflows/publish.yml (hunk 4 above)
Comment (cdavernas) [reply .github/workflows/publish.yml:28]: Added

File: .github/workflows/publish.yml (hunk 2 above)
Comment (cdavernas) [reply .github/workflows/publish.yml]: So, upon release, all projects in `./src/providers/` must be published, added to a `.tar.gz` file, then added to the release's artifacts, just like we dot for Synapse plugins

**Review by JBBianchi — APPROVED** [review]

**Pull request merged**